classes/file/FileManager.inc.php

```diff
@@ -543,21 +543,97 @@ function parseFileExtension($fileName) {
 
 	/**
 	 * Decompress passed gziped file.
-	 * @param $filePath string
-	 * @param $errorMsg string
-	 * @return boolean|string
+	 * @param $filePath string The compressed file's filename
+	 * @param $errorMsg string Any fatal error encountered (modified by reference)
+	 * @return boolean|string False on failure, or the decompressed filename on success.
 	 */
 	function decompressFile($filePath, &$errorMsg) {
+		if (extension_loaded('Zlib')) {
+			$chunk = 1024*1024;
+			$newFilePath = preg_replace('/[.]tgz$/i', '.tar.gz', $filePath);
+			$newFilePath = preg_replace('/[.]gz$/i', '', $newFilePath);
+			if ($newFilePath !== $filePath) {
+				$gz = gzopen($filePath, 'rb');
+				if ($gz) {
+					$file = fopen($newFilePath, 'wb');
```

Comment:
Feel free to declare this out-of-scope, but now that we're generating our own new filename, it's making me antsy that we're overwriting without warning. Generally I would rather us either specify a target filename as a parameter to this function, or use a function like https://www.php.net/manual/en/function.tempnam.php to safely generate a new file (returning its name as the return value, as we currently do).

Reply:
Agreed, this is terrible, but it is also the legacy function.  But the legacy function is also localized enough that we can fix it.  So, I guess will fix as well.

Reply:
So, if I fix this, I am either:
* Changing the method signature to something outrageous `...pressFile(string|array $filepathOrOldNewPathArray, array &$errorMsg)` or `...pressFile(string $oldFilePath, array &$errorMsg, string $newFilePath = null)`
* Or, changing to an incompatible but sane method signature in a minor or patch release.

Both of these are terrible.

*Muttering intensifies to grumbling.*

Reply:
Maybe cleaning this up would be best left to the `master` branch.

Reply:
I will happily pay you tomorrow for my hamburger today.

I'll fix my own goofs in this PR and write up a new PR for master (hopefully).